lib/Dialect/Seq/Transforms/LowerSeqToSV.cpp

```diff
@@ -73,10 +80,378 @@ struct CompRegLower : public OpConversionPattern<CompRegOp> {
 };
 } // namespace
 
+namespace {
+/// Lower FirRegOp to `sv.reg` and `sv.always`.
+class FirRegLower {
+public:
+  FirRegLower(hw::HWModuleOp module) : module(module), ns(module) {}
+
+  void lower();
+
+private:
+  /// Cache for constants.
+  DenseMap<APInt, Value> constants;
+
+  Value getConstant(const APInt &v) {
+    auto it = constants.try_emplace(v, Value{});
+    if (it.second) {
+      auto builder = ImplicitLocOpBuilder::atBlockBegin(module.getLoc(),
+                                                        module.getBodyBlock());
+      it.first->second = builder.create<hw::ConstantOp>(v);
+    }
+    return it.first->second;
+  }
+
+  using AsyncResetSignal = std::pair<Value, Value>;
+
+  std::pair<sv::RegOp, llvm::Optional<AsyncResetSignal>> lower(FirRegOp reg);
+
+  void initialise(OpBuilder &regBuilder, OpBuilder &initBuilder, sv::RegOp reg);
```

Comment:
LLVM coding guidelines are to put standalone functions as static outside an anonymous namespace, rather than in it.